### ISSUE TITLE
String-typed sub-queries in From-clause not parenthized

### ISSUE THREAD

**Issue by mamobyz** [codemix/oriento#198]
Statement.buildStatement() does not put the sub-query of a From-clause in parentheses when the sub-query is provided as a string, whereas a string-typed sub-query in To-clauses are parenthesized.

BTW: The sub-query syntax in From and To-clauses is identical, isn't it?

**Comment by phpnode** [comment]
@mamobyz `FROM` is a special snowflake because it's multi-purpose, consider:

```sql
SELECT FROM OUser;
```
vs
```sql
CREATE EDGE E FROM #5:1 TO #5:2;
```

The issue is that orient won't accept, e.g. this:
```sql
SELECT FROM (OUser);
```
So we need to determine whether to add parentheses by looking at the string. I guess the most reliable way is to check whether it contains spaces.

**Comment by phpnode** [comment]
The broader issue is that the query builder is not context aware. this will be fixed in the next major version

**Comment by mamobyz** [comment]
@phpnode: Thanks for the quick reply!
Yes, checking white spaces would be a solution, just as To-clause does.

Regarding lacking of context awareness, maybe renaming from() and to() methods to out() and in() for building query/statement for edge creations in oriento is a solution. E.g.:
     db.create('edge', E).out(string || function).in(string || function)
Of course, it does not adhere to the intention of keeping the method names identical to the SQL clauses.

**Comment by phpnode** [comment]
yeah for the query builder I'd prefer to keep the names the same as the sql keywords. I've added a fix in #200 that wraps queries in parens if they're not already wrapped *and* they contain spaces.

Regarding context, the next version of the query builder will use different object classes, so the `.from()` in `db.select().from('Something')` will not be the same as in `db.create().edge('E').from('Something')`
